agent/xds/jwt_authn_test.go

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package xds
+
+import (
+	"path/filepath"
+	"testing"
+
+	"github.com/hashicorp/consul/agent/structs"
+	"github.com/stretchr/testify/require"
+)
+
+func TestMakeJWTAUTHFilters(t *testing.T) {
```

Comment:
I think this test can be improved in a few ways:
1. All test cases should use one `pCE` containing all providers. The first parameter to `makeJWTAuthFilter` is in practice a lookup table of all known providers. I don't think there's value in separating `remoteCE` and `localCE` since we really care about how intentions get transformed.
2. We should test multiple intentions. 
3. We should test error scenarios like `provider specified in intention does not exist` or other misconfigurations

Reply:
note: We commented about additional tests cases in the enterprise PR, and agreed to include those in the follow up where `VerifyClaims` is supported (which is going in for 1.16).